pkg/controllers/machine/disruption/drift.go

```diff
@@ -83,12 +85,30 @@ func (d *Drift) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisioner
 // isDrifted will check if a machine is drifted from the fields in the provisioner.Spec and
 // the cloudprovider
 func (d *Drift) isDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (bool, error) {
```

Comment:
Is there a way to generalize this so we don't have to code this in so specifically?

Reply:
We should also be thinking about how much the user cares about the kind of drift that caused the disruption. Do we think (at scale) this is relevant information to the user? Is there a way that we could codify the type of drift through the disruption reason or message in some way?